runtime/common/src/migrations/nuke.rs

```diff
@@ -24,19 +24,93 @@ use sp_runtime::DispatchError;
 #[cfg(feature = "try-runtime")]
 use sp_std::vec::Vec;
 
+/// This upgrade nukes all storage from the pallet individually.
+///
+/// If the pallet shall be kept, please use [ResetPallet] instead because
+/// here we neither check nor reset the corresponding storage version.
+pub struct KillPrefix<PalletName, DbWeight>(sp_std::marker::PhantomData<(PalletName, DbWeight)>);
```

Comment:
NOTE: I tried reusing the existing `ResetPallet` migration. Unfortunately this requires the pallet to still be defined for the runtime which we don't want. The solution would have been super hacky, so I decided to create a separate migration which nukes a pallet based on the pallet name.